app/services/katello/pulp3/smart_proxy_mirror_repository.rb

```diff
@@ -37,6 +37,58 @@ def orphan_repository_versions
         repo_version_map
       end
 
+      def report_misconfigured_repository_version(api, href)
+        # Reasons for distributions distributing orphaned repository versions:
+        # 1. The sync succeeded but Pulp did not update the publication (yum content)
+        #    - Fix: completely resync the repository to the smart proxy (need to verify)
+        # 2. The sync suceeded but metadata was not generated (non-yum content)
+        #    - Fix: completely resync the repository on the smart proxy (need to verify)
+        # 3. A repository, distribution, and publication was lost track of
+        #    - Fix: same as 4
+        # 4. Pulp content was modified outside of Katello
+        #    - Fix: find repositories outside of Katello and delete them. Deleting the entire repo works and leaves an orphaned distribution.
+        #        - If RemoveUnneededRepos goes first, this should be taken care of.
```

Comment:
I think there's a gap here if a repo was versioned, published and distributed outside of katello. self.orphan_distribution? doesn't catch that case. Not sure how to handle that..Automatically would be checking if disribution belongs to a version that is orphaned.

Reply:
In this case we can detect and delete the orphaned distribution by comparing distribution names to pulp_id on repository records.